bundles/org.openhab.binding.cloudrain/src/main/java/org/openhab/binding/cloudrain/internal/handler/CloudrainZoneHandler.java

```diff
@@ -0,0 +1,427 @@
+     */
+    public CloudrainZoneHandler(Thing thing, CloudrainAPI cloudrainAPI, TimeZoneProvider timeZoneProvider,
+            ItemChannelLinkRegistry itemChannelLinkRegistry) {
+        super(thing);
+        this.cloudrainAPI = cloudrainAPI;
+        this.timeZoneProvider = timeZoneProvider;
+        this.itemChannelLinkRegistry = itemChannelLinkRegistry;
+    }
+
+    @Override
+    public void initialize() {
+        // Retrieve this zone's ID from the properties as set by discovery
+        String zoneIdFromProperty = getThing().getProperties().get(PROPERTY_ZONE_ID);
+        if (zoneIdFromProperty != null) {
+            zoneId = zoneIdFromProperty;
+            // schedule initialization tasks
+            scheduler.execute(() -> {
+                CloudrainAccountHandler handler = getAccountHandler();
+                if (handler != null) {
+                    // Register the zone to receive property and status updates
+                    handler.registerZone(zoneId, getThing());
+                    // Call the Cloudrain API and update the thing's status now
+                    boolean successful = updateStatusFromAPI();
+                    if (successful) {
+                        // Register the zone to receive irrigation updates only if channels are linked
+                        // This is done to avoid unnecessary frequent API calls
+                        if (isAnyIrrigationChannelLinked()) {
+                            handler.registerForIrrigationUpdates(zoneId, getThing());
+                        }
+                        // Update the THING status
+                        updateStatus(ThingStatus.ONLINE);
+                    }
+                } else {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.HANDLER_MISSING_ERROR,
+                            String.format(ERROR_MSG_ZONE_REGISTRATION, zoneId));
+                }
+            });
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, ERROR_MSG_ZONE_ID_PROPERTY);
+        }
+    }
+
+    @Override
+    public void dispose() {
+        super.dispose();
+        // unregister this zone from the account handler
+        unregisterAll();
+    }
+
+    @Override
+    public void handleRemoval() {
+        super.handleRemoval();
+        unregisterAll();
+    }
+
+    private void unregisterAll() {
+        // unregister this zone from the account handler
+        CloudrainAccountHandler handler = getAccountHandler();
+        if (handler != null) {
+            handler.unregisterZone(zoneId);
+            handler.unregisterFromIrrigationUpdates(zoneId);
+        }
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        // handle irrigation commands
+        if (CHANNEL_GROUP_ID_COMMAND.equals(channelUID.getGroupId())) {
+            if (command instanceof OnOffType && ((OnOffType) command) == OnOffType.ON) {
```

Comment:
No need to check instanceof with an Enum.
```suggestion
            if (command == OnOffType.ON) {
```

Reply:
Fixed